axum/src/docs/routing/nest.md

```diff
@@ -103,18 +73,79 @@ the prefix stripped:
 ```rust
 use axum::{routing::get, http::Uri, Router};
 
+let nested_router = Router::new()
+    .route("/", get(|uri: Uri| async {
+        // `uri` will _not_ contain `/bar`
+    }));
+
 let app = Router::new()
     .route("/foo/*rest", get(|uri: Uri| async {
         // `uri` will contain `/foo`
     }))
-    .nest("/bar", get(|uri: Uri| async {
-        // `uri` will _not_ contain `/bar`
-    }));
+    .nest("/bar", nested_router);
 # async {
 # axum::Server::bind(&"".parse().unwrap()).serve(app.into_make_service()).await.unwrap();
 # };
 ```
 
+# Differences between `nest` and `nest_service`
+
+When [fallbacks] are called differs between `nest` and `nested_service`. Routers
+nested with `nest` will delegate to the outer router's fallback if they don't
+have a matching route, whereas `nested_service` will not.
```

Comment:
This makes it sound like nested routers won't call their own fallback if one is configured. I'm hoping it really means "if the nested router does not have an explicitly-configured fallback".

For perspective, as a user, a fallback is not a "route", and routers don't have "default fallbacks", routers are just collections of routes with an optional fallback, and there's some default behavior in the case where no fallback is configured.

Reply:
A service nested via `Router::nest_service` will not call so the outer routers fallback. If the request matches the prefix then it goes into the opaque nested service and doesn't leave again. This is because axum cannot tell whether the response from the service was from a route or a fallback.

If you need that you gotta use `Router::nest`, which doesn't allow opaque services and instead only `Router`s.

With https://github.com/tokio-rs/axum/pull/1161 we'll support fallbacks on nested routers so I'm not sure there is much reason to call `nest_service` with a `Router`.

Reply:
I'm talking specifically about what this says about `Router::nest`. It says "Routers nested with `nest` will delegate to the outer router's fallback if they don't have a matching route". I suppose that with this PR alone nested routers don't allow fallbacks so this paragraph is okay, but #1161 leaves this language intact.

Given that, I guess #1161 needs to update these docs to better explain fallback behavior with `nest` (and to fix the example of adding a fallback to the nested router as it's currently done via `nest_service`).

Reply:
Yep that's right. 1161 is still a draft and has a todo for updating the docs.